include/inexor/vulkan-renderer/input/keyboard_mouse_data.hpp

```diff
@@ -26,55 +28,52 @@ class KeyboardMouseInputData {
     KeyboardMouseInputData &operator=(KeyboardMouseInputData &&) = delete;
 
     /// @brief Change the key's state to pressed.
-    /// @param key The key which was pressed.
+    /// @param key the key which was pressed
     void press_key(std::int32_t key);
 
     /// @brief Change the key's state to unpressed.
-    /// @param key The key which was released.
+    /// @param key the key which was released
     void release_key(std::int32_t key);
 
+    /// @brief Check if the given key is currently pressed.
+    /// @param key the key index
+    /// @return ``true`` if the key is pressed
+    [[nodiscard]] bool is_key_pressed(std::int32_t key) const;
+
+    /// @brief Checks if a key was pressed once.
+    /// @param key The key index
+    /// @return ``true`` if the key was pressed
+    [[nodiscard]] bool was_key_pressed_once(std::int32_t key);
+
     /// @brief Change the mouse button's state to pressed.
-    /// @param button The mouse button which was pressed.
+    /// @param button the mouse button which was pressed
     void press_mouse_button(std::int32_t button);
 
     /// @brief Change the mouse button's state to unpressed.
-    /// @param button The mouse button which was released.
+    /// @param button the mouse button which was released
     void release_mouse_button(std::int32_t button);
 
-    /// @brief Set the current cursor position.
-    /// @param cursor_pos_x The current x-coordinate of the cursor.
-    /// @param cursor_pos_y The current y-coordinate of the cursor.
-    void set_cursor_pos(double cursor_pos_x, double cursor_pos_y);
+    /// @brief Check if the given mouse button is currently pressed.
+    /// @param button the mouse button index
+    /// @return ``true`` if the mouse button is pressed
+    [[nodiscard]] bool is_mouse_button_pressed(std::int32_t button) const;
 
-    /// @return Return a std::array of size 2 which contains the x-position in index 0 and y-position in index 1.
-    [[nodiscard]] std::array<std::int64_t, 2> get_cursor_pos() const;
+    /// @brief Checks if a mouse button was pressed once.
+    /// @param button the mouse button index
+    /// @return ``true`` if the mouse button was pressed
+    [[nodiscard]] bool was_mouse_button_pressed_once(std::int32_t button);
```

Comment:
Document that if `button` is to high it's undefined behaviour?

Reply:
Pretty sure that `std::array::operator[]` doesn't have any assertions, so it might be wise to add an `assert` ourselves.

Reply:
I added `assert` and doxygen comments for documentation. Done.